core/client/fs/src/main/java/alluxio/client/file/cache/store/MemoryPageStore.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.client.file.cache.store;
+
+import alluxio.client.file.cache.PageId;
+import alluxio.client.file.cache.PageInfo;
+import alluxio.client.file.cache.PageStore;
+import alluxio.exception.PageNotFoundException;
+import alluxio.exception.status.ResourceExhaustedException;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.concurrent.NotThreadSafe;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Stream;
+
+/**
+ * The {@link MemoryPageStore} is an implementation of {@link PageStore} which
+ * stores all pages in memory.
+ */
+@NotThreadSafe
+public class MemoryPageStore implements PageStore {
+  private static final Logger LOG = LoggerFactory.getLogger(MemoryPageStore.class);
+  private final long mPageSize;
+  private final long mCapacity;
+  private final int mFileBuckets;
+
+  private ConcurrentHashMap<String, byte[]> mPageStoreMap = null;
+
+  /**
+   * Creates a new instance of {@link MemoryPageStore}.
+   *
+   * @param options options for the buffer store
+   */
+  public MemoryPageStore(MemoryPageStoreOptions options) {
+    mPageSize = options.getPageSize();
+    mCapacity = (long) (options.getCacheSize() / (1 + options.getOverheadRatio()));
+    mFileBuckets = options.getFileBuckets();
+    mPageStoreMap = new ConcurrentHashMap<>();
+  }
+
+  @Override
+  public void put(PageId pageId, byte[] page) throws ResourceExhaustedException, IOException {
+    String pageKey = getKeyFromPageId(pageId);
+    try {
+      byte[] mPage = new byte[page.length];
+      System.arraycopy(page, 0, mPage, 0, page.length);
```

Comment:
is the arraycopy here necessary?  I think when we read a page from external file system, we already made a copy there.

Reply:
I think it's necessary, if use the byte array data directly, when the original byte array data is modified, the value is changed too.

Reply:
hmmm, do you happen to know when and how the original btye array could be modified?  Thanks!

Reply:
For example:
```
        InputStream is = new FileInputStream("XXXX/alluxio/NOTICE");
        ByteArrayOutputStream buffer = new ByteArrayOutputStream();

        int nRead;
        long cnt = 0;
        byte[] data = new byte[4];   // read buffer

        while ((nRead = is.read(data, 0, data.length)) != -1) {
            // put data into memory cache
            mPageDataMap.put(cnt, data);
            cnt++;
            buffer.write(data, 0, nRead);
        }

        buffer.flush();
```
Use the byte array as a buffer, read data from a file and put the data into memory cache, at the end, all the cached data is the last one read from the file.